TwitchPlaysAssembly/Src/ComponentSolvers/ComponentSolver.cs

```diff
@@ -1131,40 +1064,10 @@ private void AwardStrikes(string userNickName, int strikeCount)
 		if (currentReward != originalReward)
 			messageParts.Add($"Reward {(currentReward > 0 ? "reduced" : "increased")} to {currentReward} points.");
 
-		if (OtherModes.VSModeOn)
+		if (VSAffect)
 		{
-			if (userNickName != null)
-			{
-				if (hpPenalty != 0)
-				{
-					if (team == OtherModes.Team.Good)
-						OtherModes.SubtractGoodHealth(hpPenalty);
-					else
-						OtherModes.SubtractEvilHealth(hpPenalty);
-				}
-				else
-				{
-					if (team == OtherModes.Team.Evil)
-					{
-						OtherModes.evilHealth = 0;
-
-						// If Evil loses, the bomb is solved.
-						GameCommands.SolveBomb();
-					}
-					else
-					{
-						// If Good loses, the bomb blows up.
-						TwitchPlaysService.Instance.CoroutineQueue.CancelFutureSubcoroutines();
-						TwitchGame.Instance.Bombs[0].CauseVersusExplosion();
-
-						// This was here originally to detonate the bomb, but was nonfunctional.
-						//TwitchPlaysService.Instance.CoroutineQueue.AddToQueue(BombCommands.Explode(TwitchGame.Instance.Bombs[0]));
-					}
-				}
-
-				if (hpPenalty != 0)
-					TwitchGame.ModuleCameras.UpdateConfidence();
-			}
+			CalculateVSHP(userNickName, strikePenalty, out OtherModes.Team? teamDamaged, out int HPDamage); //can't run it outside due to userNickName possibly being empty, I don't like running it again but it's the most efficient way
```

Comment:
Sorry, I'm a bit confused. Why is CalculateVSHP being called on this line and 1038?

Reply:
Visual Studio was telling me that the variables `out OtherModes.Team? teamDamaged, out int HPDamage` weren't already declared, and the previous call for `CalculateVSHP()` couldn't go outside of the if statement because `userNickName` could have been empty. I don't think anything could change from the first to the second call of `CalculateVSHP()`, so although I don't like repeating this, it is the shortest solution.

Reply:
Oh, I understand now. You can make the variables outside of the first if statement and just do `out teamDamaged, out HPDamage`. I also wonder if we could call `VSUpdate(teamDamaged, HPDamage)` in the first if statement instead of calling it later. I'm not sure why it's called later.

Reply:
Ok, that change is made. It's ordered the way it is to make sure that a solve / strike is awarded and announced in chat before the bomb could potentially solve or detonate.